core/src/dird/backup.cc

```diff
@@ -949,12 +949,12 @@ void GenerateBackupSummary(JobControlRecord *jcr, ClientDbRecord *cr, int msg_ty
       bstrncpy(compress, "None", sizeof(compress));
    } else {
       compression = (double)100 - 100.0 * ((double)jcr->JobBytes / (double)jcr->ReadBytes);
-      if (compression < 0.5) {
-         bstrncpy(compress, "None", sizeof(compress));
-      } else {
-         Bsnprintf(compress, sizeof(compress), "%.1f %%", compression);
-         FindUsedCompressalgos(&compress_algo_list, jcr);
+      if(compression < 0){
+        Jmsg(jcr, M_WARNING, 0, _("Negative compression rate detected! "
+              "Disable compression for this fileset or choose different algorithm.\n"));
```

Comment:
I would advise against this warning.  Think about the case were you do an incremental after only very few changes to incompressible files (for example you just updated some system libraries).  This probably trigger this warning even though the full fileset (which in this example is the whole filesystem) might be very compressible.

Reply:
I have changed this in the last commit, so that we only get an INFO if a full backup expanded by more than 1%